File: exp-log.py
import numpy as np

#new exponential definition
def my_exp(x,u):
  #u: umbral
    if  x < u and x >= -u:
      return np.exp(x)
    elif x < -u:
      return np.exp(-u)
    else:
      return np.exp(u)

#new logarithm definition
def my_ln(x,u):
  #u: umbral
    if x < u:
      return np.log(u)
    else:
      return np.log(x)

File: exp-log.R
my_ln<-function(x,u){
    #u: umbral
    y<-ifelse(x<u,u,x)
    return(log(y))
  }
  
  #Definicipon de nuevo exp
  my_exp<-function(x,u){
    #u: umbral
    y<-ifelse(x>=(-u) & x<u, x, ifelse(x<(-u),-u,u))
    return(exp(y))
  }